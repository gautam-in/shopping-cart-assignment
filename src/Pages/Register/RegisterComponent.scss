/******************* RegisterComponent *****************************/
@import '../../Styles/variable.scss';
@import '../../Styles/mixin.scss';

.register {
    width: 60%;
    margin: $LARGE_MARGIN auto;
    @include grid-2-column;
    gap: $SMALL_GAP;

    & .register_page-header {
        @include flex-column;
        gap: $MEDIUM_GAP;

        & .register_page-header_heading {
            font-size: $LARGE_FONT_SIZE;
        }

        & .register_page-header_subsection {
            font-size: $MEDIUM_FONT_SIZE;
        }

    }

    & .register_page-content {
        @include flex-column;
        gap: $LARGE_GAP;
        justify-content: space-evenly;

        & .register_page-content_button {
            margin-top: $SMALL_MARGIN;
        }
    }
}