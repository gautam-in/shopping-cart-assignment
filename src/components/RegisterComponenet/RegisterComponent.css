.signup__container {
  display: flex;
  justify-content: space-around;
  padding-top: 3rem;
  padding-bottom: 1.5rem;
  max-width: 1024;
  margin: 0 auto;
}
@media screen and (max-width: 480px) {
  .signup__container {
    flex-direction: column;
  }
}
.signup-container__article {
  flex: 4;
  display: flex;
  flex-direction: column;
  padding: 0 0rem;
  align-items: center;
}
@media screen and (max-width: 480px) {
  .signup-container__article {
    padding-bottom: 3rem;
    padding-left: 1rem;
  }
}
.signup-container__form {
  flex: 5;
}