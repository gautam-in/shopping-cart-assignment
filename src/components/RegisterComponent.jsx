import React from "react";
import "./LoginComponent.css";
import { useHistory } from "react-router-dom";
import { Formik } from "formik";
import * as Yup from "yup";
function RegisterComponent() {
  const history = useHistory();

  return (
    <section className="signup-section">
      <div className="signup-section_info">
        <h1 className="login-section-heading">Signup</h1>
        <div className="login-section-desc">
          We do nto share your personal details anywhere.
        </div>
      </div>
      <div className="login-section_formGroup">
        <Formik
          initialValues={{
            firstName: "",
            lastName: "",
            email: "",
            password: "",
            confirmPassword: "",
          }}
          onSubmit={(values, { setSubmitting }) => {
            setTimeout(() => {
              console.log("Logging in", values);
              setSubmitting(false);
              history.push(`/`);
            }, 500);
          }}
          validationSchema={Yup.object().shape({
            firstName: Yup.string()
              .required("First Name is required")
              .min(3, "First Name is too short - should be 3 chars minimum."),
            lastName: Yup.string()
              .required("Last Name is required")
              .min(1, "Last Name is too short - should be 1 chars minimum."),
            email: Yup.string().email().required("Email is required"),
            password: Yup.string()
              .required("Password is required")
              .min(6, "Password is too short - should be 6 chars minimum.")
              .matches(/(?=.*[0-9])/, "Password must contain a number."),
            confirmPassword: Yup.string()
              .required("Confirm Password is required")
              .min(
                6,
                "Confirm Password is too short - should be 6 chars minimum."
              )
              .matches(/(?=.*[0-9])/, "Password must contain a number.")
              .oneOf([Yup.ref("password"), null], "Passwords must match"),
          })}
        >
          {(props) => {
            const {
              values,
              touched,
              errors,
              isSubmitting,
              handleChange,
              handleBlur,
              handleSubmit,
            } = props;
            return (
              <form
                onSubmit={handleSubmit}
                autoComplete="off"
                className="signup-section_form"
                aria-label="form"
              >
                <div className="field">
                  <input
                    type="text"
                    name="firstName"
                    id="firstName"
                    placeholder="First Name"
                    value={values.firstName}
                    onChange={handleChange}
                    onBlur={handleBlur}
                  />
                  <label htmlFor="firstName" id="firstName">
                    First Name *
                  </label>
                </div>
                {errors.firstName && touched.firstName && (
                  <span style={{ color: "red" }}>{errors.firstName}</span>
                )}
                <div className="field">
                  <input
                    type="text"
                    name="lastName"
                    id="lastName"
                    placeholder="Last Name"
                    placeholder="Enter your email"
                    value={values.lastName}
                    onChange={handleChange}
                    onBlur={handleBlur}
                  />
                  <label htmlFor="lastName" id="lastName">
                    Last Name*
                  </label>
                </div>
                {errors.lastName && touched.lastName && (
                  <span style={{ color: "red" }}>{errors.lastName}</span>
                )}
                <div className="field">
                  <input
                    name="email"
                    id="email"
                    type="text"
                    placeholder="Enter your email"
                    value={values.email}
                    onChange={handleChange}
                    onBlur={handleBlur}
                  />
                  <label htmlFor="email" id="email">
                    Email *
                  </label>
                </div>
                {errors.email && touched.email && (
                  <span style={{ color: "red" }}>{errors.email}</span>
                )}
                <div className="field">
                  <input
                    type="password"
                    name="password"
                    id="password"
                    placeholder="Enter your password"
                    value={values.password}
                    onChange={handleChange}
                    onBlur={handleBlur}
                  />
                  <label htmlFor="password" id="password">
                    Password *
                  </label>
                </div>
                {errors.password && touched.password && (
                  <span style={{ color: "red" }}>{errors.password}</span>
                )}

                <div className="field">
                  <input
                    type="password"
                    name="confirmPassword"
                    id="confirmPassword"
                    placeholder="Confirm Password"
                    value={values.confirmPassword}
                    onChange={handleChange}
                    onBlur={handleBlur}
                  />
                  <label htmlFor="confirmPassword" id="confirmPassword">
                    Confirm Password *
                  </label>
                </div>
                {errors.confirmPassword && touched.confirmPassword && (
                  <span style={{ color: "red" }}>{errors.confirmPassword}</span>
                )}
                <input
                  type="submit"
                  className="btn--signup"
                  disabled={isSubmitting}
                  name="submit"
                  value="SignUp"
                />
              </form>
            );
          }}
        </Formik>
      </div>
    </section>
  );
}

export default RegisterComponent;
