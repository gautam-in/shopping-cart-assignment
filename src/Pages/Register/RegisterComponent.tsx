import { FormEvent } from "react";

import "./RegisterComponent.scss";
import { CONSTANT } from "../../Utility/Constant";
import { InputSubmitButtonComponent } from "../../BaseComponent/InputButton/InputSubmitButtonComponent";
import { InputTextBoxComponent } from "../../BaseComponent/InputTextBox/InputTextBoxComponent";

export const RegisterComponent: React.FC = () => {
  const BASE_CLASSNAME: string = "register";
  const PAGE_HEADER_CLASSNAME: string = `${BASE_CLASSNAME}_page-header`;
  const HEADER_HEADING_CLASSNAME: string = `${PAGE_HEADER_CLASSNAME}_heading`;
  const HEADER_SUBSECTION_CLASSNAME: string = `${PAGE_HEADER_CLASSNAME}_subsection`;
  const PAGE_CONTENT_CLASSNAME: string = `${BASE_CLASSNAME}_page-content`;
  const BUTTON_PAGE_CONTENT_CLASSNAME: string = `${PAGE_CONTENT_CLASSNAME}_button`;

  const handleFormSubmit = (e: FormEvent<HTMLFormElement>) => {
    e.preventDefault();
    console.log("Form Subbmitted");
  };

  return (
    <div className={BASE_CLASSNAME}>
      <div className={PAGE_HEADER_CLASSNAME}>
        <h1 className={HEADER_HEADING_CLASSNAME}>
          {CONSTANT.REGISTER_PAGE_LOGIN_HEADER}
        </h1>
        <p className={HEADER_SUBSECTION_CLASSNAME}>
          {CONSTANT.REGISTER_PAGE_HEADER_SUBSECTION}
        </p>
      </div>
      <form onSubmit={handleFormSubmit} className={PAGE_CONTENT_CLASSNAME}>
      <InputTextBoxComponent
          id="registerFirstName"
          type="text"
          name="firstName"
          labelAndPlaceholder="First Name"
        />
        <InputTextBoxComponent
          id="registerLastName"
          type="text"
          name="lastName"
          labelAndPlaceholder="Last Name"
        />
        <InputTextBoxComponent
          id="registerEmail"
          type="email"
          name="email"
          labelAndPlaceholder="Email"
        />
        <InputTextBoxComponent
          id="registerPassword"
          type="password"
          name="password"
          labelAndPlaceholder="Password"
        />
        <InputTextBoxComponent
          id="registerConfirmPassword"
          type="password"
          name="confirmPassword"
          labelAndPlaceholder="Confirm Password"
        />
        <InputSubmitButtonComponent
          id="registerSignup"
          buttonName="Signup"
          externalClassName={BUTTON_PAGE_CONTENT_CLASSNAME}
        />
      </form>
    </div>
  );
};
