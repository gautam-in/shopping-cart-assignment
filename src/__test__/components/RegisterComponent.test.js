import { shallow,configure } from "enzyme";
import Adapter from 'enzyme-adapter-react-16';
import RegisterComponent from '../../components/RegisterComponent';
  
configure({ adapter: new Adapter() })

describe('RegisterComponent rendering', () => {
    it('should render our Snapshots correctly', () => {
        const wrapper = shallow(<RegisterComponent />);
        expect(wrapper).toMatchSnapshot();
    });
});