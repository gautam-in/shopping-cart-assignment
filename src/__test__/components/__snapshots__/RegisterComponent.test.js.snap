// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterComponent rendering should render our Snapshots correctly 1`] = `ShallowWrapper {}`;
